modules/geolocation/geolocationWorker.py

```diff
@@ -0,0 +1,89 @@
+from modules.geolocation.geolocation import Geolocation
+import multiprocessing as mp
+# For exception handling
+import queue
+
+
+def exit_requested(requestQueue):
+    return not requestQueue.empty()
+
+
+def geolocation_worker(pause, exitRequest, pipelineIn, pipelineOut, pipelineOutLock):
+
+    print("Start Geolocation")
+    locator = Geolocation()
+
+    while True:
+
+        pause.acquire()
+        pause.release()
+
+        # Pixel coordinates of tents and plane data
+        coordinates = pipelineIn.get()
+        # TODO verify this
+        if (not coordinates):
+            continue
+
+        ret, location = locator.run_locator(coordinates)
+
+        # Something has gone wrong, skip
+        if (not ret):
+            continue
+
+        # External lock required as geolocationOutput will dump the queue
+        pipelineOutLock.acquire()
+        pipelineOut.put(location)
+        pipelineOutLock.release()
+
+        if (exit_requested(exitRequest)):
+            break
+
+
+    print("Stop Geolocation")
+    return
+
+
+def geolocation_output(pause, exitRequest, pipelineIn, pipelineOut, pipelineInLock):
+
+    print("Start Geolocation Output")
+    locator = Geolocation()
+
+    while True:
+
+        pause.acquire()
+        pause.release()
+
+        # Dump the contents of the input queue
+        locations = []
+        pipelineInLock.acquire()
+        while True:
+
+            try:
+
+                location = pipelineIn.get_nowait()
```

Comment:
Note that this will raise an Empty exception if the pipeline is empty. I'm guessing the goal with this is to prevent the queue from ever being empty using the pipelineLock? Then get and get_nowait would have no real difference, the only difference occurs if the queue is empty and if we do intend for there to be a case where the queue is intentionally empty, we should have exception handling for the Empty exception. If not, then I don't know why get_nowait() is being used.

Reply:
No, we're dumping the entire queue and we don't want to block on empty.

Reply:
Got it, in that case you can either check if its empty or just have a try catch block, just because this will throw an error once the entire queue has been extracted and the last one is left.

Reply:
Oh lmao already done, my bad